### PR TITLE
Line Chart - update and fix context

### PR DESCRIPTION
**Pull request by PrimozGodec**
##### Issue

Context didn't correctly store and update settings when features were at the input.

##### Description of changes
This pull request implements a custom context handler that separately handles features on the input and cases where features are not present.
It fixes: https://github.com/biolab/orange3-timeseries/issues/146, https://github.com/biolab/orange3-timeseries/issues/146, https://github.com/biolab/orange3-timeseries/issues/160
I hope it solves: https://github.com/biolab/orange3-timeseries/issues/166, https://github.com/biolab/orange3-timeseries/issues/161 (cannot reproduce those errors and do not see complete message)

##### Includes
- [X] Code changes
- [ ] Tests
- [ ] Documentation

### CODE REVIEW

File: orangecontrib/timeseries/widgets/owlinechart.py [hunk 1]
```diff
@@ -405,40 +411,47 @@ def setType(self, ax, type):
...
+            if hasattr(context, "features") and context.features == features:
+                return self.PERFECT_MATCH
+            else:
+                return self.NO_MATCH
+        else:
+            # match if all selected attributes in domain
+            # when no features on input match just contexts with no features
+            values = set(y for x in attrs for y in x)
+            if (
+                context.features is None and len(values) > 0
```
Comment (VesnaT) [inline orangecontrib/timeseries/widgets/owlinechart.py:449]: At this point the `context` may not have the `features` attribute.

One way of handling it, is to implement the `migrate_context`.

File: orangecontrib/timeseries/widgets/owlinechart.py [hunk 2]
```diff
@@ -405,40 +411,47 @@ def setType(self, ax, type):
...
 
 
-class LineChartContextHandler(DomainContextHandler):
+class LineChartContextHandler(ContextHandler):
     """
     Context handler of line chart. The specifics of this widget is
     that `attrs` variable is list of lists and it is not handled with
     the DomainContextHandler.
     """
+    def open_context(self, widget, domain_fetures):
```
Comment (VesnaT) [inline orangecontrib/timeseries/widgets/owlinechart.py:420]: domain_fetures -> domain_features

File: orangecontrib/timeseries/widgets/owlinechart.py [hunk 3]
```diff
@@ -405,40 +411,47 @@ def setType(self, ax, type):
...
+        if domain_fetures[0] is None:
```
Comment (VesnaT) [inline orangecontrib/timeseries/widgets/owlinechart.py:421]: Could this ever be True? As far as I see, the context is opened only when data is not None.

**Review by VesnaT — COMMENTED** [review]
When I save the following case to a workflow and open it, the settings are set to defaults (instead of as show on the image).
image
I suppose the 'saving workflow' has an issue, since re-placing the widget on the canvas works fine.

File: orangecontrib/timeseries/widgets/owlinechart.py (hunk 1 above)
Comment (PrimozGodec) [reply orangecontrib/timeseries/widgets/owlinechart.py:443]: Didn't think about that. Thank you. :) Now handler is even simpler.

File: orangecontrib/timeseries/widgets/owlinechart.py (hunk 3 above)
Comment (PrimozGodec) [reply orangecontrib/timeseries/widgets/owlinechart.py]: Thank you. It was left there from the previous implementation which was different. Reimplementing open_context is not required anymore. I removed this funciton.

**Comment by PrimozGodec** [comment]
Now I use `storeSpecificSettings` so contexts should be saved to file correctly. :)

**Comment by VesnaT** [comment]
This probably fixes #171 too.

**Pull request merged**